File: program/find-smallest-element-in-a-matrix/find-smallest-element-in-a-matrix.py
# smallest element in a matrix


def findSmallestElementInAMatrix(matrix):

    smallestElement = min(matrix[0])

    for i in range(1,len(matrix)):
        if min(matrix[i]) < smallestElement:
            smallestElement = min(matrix[i])

    
    return smallestElement


if __name__=="__main__":
    print(findSmallestElementInAMatrix([[10,222,43],[0,-15,336],[437,854,49]]))

File: program/find-smallest-element-in-a-matrix/find_smallest_element_in_a_matrix.dart
void smallestInMatrix(List<List<int>> mat){
  int smallest = mat[0][0];
  for(var i = 0 ; i < mat.length ; i++){
    for(var j = 0 ; j < mat[i].length ; j++){
      if(mat[i][j] < smallest){
        smallest = mat[i][j];
      }
    }
  }
  print(smallest);
}

void main() {
  smallestInMatrix([[1,2,3,4,5],[0,-1,-6,-9,0]]);
}